manager/src/main/java/org/openremote/manager/datapoint/AssetDatapointService.java

```diff
@@ -233,20 +245,51 @@ protected String buildWhereClause(List<Pair<String, Attribute<?>>> attributes, b
     public ScheduledFuture<File> exportDatapoints(AttributeRef[] attributeRefs,
                                                   long fromTimestamp,
                                                   long toTimestamp) {
+        try {
+            String query = getSelectExportQuery(attributeRefs, fromTimestamp, toTimestamp);
+
+            // Verify the query is 'legal' and can be executed
+            if(canQueryDatapoints(query, null, datapointExportLimit)) {
+                return doExportDatapoints(attributeRefs, fromTimestamp, toTimestamp);
+            }
+            throw new RuntimeException("Could not export datapoints.");
+
+        } catch (DatapointQueryTooLargeException dqex) {
+            String msg = "Could not export data points. It exceeds the data limit of " + datapointExportLimit + " data points.";
+            getLogger().log(Level.WARNING, msg, dqex);
+            throw dqex;
+        }
+    }
+
+    protected ScheduledFuture<File> doExportDatapoints(AttributeRef[] attributeRefs,
+                                                       long fromTimestamp,
+                                                       long toTimestamp) {
+
         return scheduledExecutorService.schedule(() -> {
             String fileName = UniqueIdentifierGenerator.generateId() + ".csv";
-            StringBuilder sb = new StringBuilder(String.format("copy (select ad.timestamp, a.name, ad.attribute_name, value from asset_datapoint ad, asset a where ad.entity_id = a.id and ad.timestamp >= to_timestamp(%d) and ad.timestamp <= to_timestamp(%d) and (", fromTimestamp / 1000, toTimestamp / 1000))
-                .append(Arrays.stream(attributeRefs).map(attributeRef -> String.format("(ad.entity_id = '%s' and ad.attribute_name = '%s')", attributeRef.getId(), attributeRef.getName())).collect(Collectors.joining(" or ")))
-                .append(")) to '/storage/")
-                .append(EXPORT_STORAGE_DIR_NAME)
-                .append("/")
-                .append(fileName)
-                .append("' delimiter ',' CSV HEADER;");
+            StringBuilder sb = new StringBuilder(String.format("copy ("))
```

Comment:
Redundant String formatting:

```suggestion
            StringBuilder sb = new StringBuilder("copy (")
```